python/cudf_polars/cudf_polars/experimental/rapidsmpf/frontend/spmd.py

```
@@ -336,6 +336,11 @@ def __init__(
         executor_options: dict[str, Any] | None = None,
         engine_options: dict[str, Any] | None = None,
     ) -> None:
+        from cudf_polars.experimental.rapidsmpf.frontend.default_singleton_engine import (
+            check_no_live_default_singleton,
+        )
```

Ruff (0.15.12)

[warning] `import` should be at the top-level of a file

(PLC0415)